lib/circuitbox/errors/open_circuit_error.rb

```diff
@@ -5,7 +5,7 @@ class OpenCircuitError < Circuitbox::Error
     attr_reader :service
 
     def initialize(service)
-      super()
+      super("Service #{service.inspect} has an open circuit")
```

Comment:
Since the service is converted into a string in `CircuitBreaker` is there a specific reason inspect is being called?

I think it makes sense to have some standard exception message for this error. Previously one would need to rescue the exception, read the value of service, and construct their own message. This gets rid of the need to do that but with the downside of always allocating the message regardless of it being read. The message should be lazily created by overriding the `to_s` method like what `ServiceFailureError` does.

Reply:
`inspect` is to make the substituted value separate from the surrounding text for readability. Imagine you are using local hostnames for your services, for example you have a machine called `gollum` and you can reach it with its' bare hostname. If you template the name into a message "as is" it can be confusing because your message will look like `Service gollum has an open circuit`. If the service name consists of multiple words separated by spaces the value visually dissolves in the rest of the message. Since there are no strictly defined semantics for quoting / Markdown in exception messages I opt for the next best thing and use `inspect` for this  "auto-quote". I think in some other instances a combo of backtick+single quote is used too in some Ruby builtin messages, we can also uses that?